pytorch_lightning/callbacks/pt_callbacks.py

```diff
@@ -30,40 +31,55 @@ def set_model(self, model):
             model = model.module
         self.model = model
 
-    def on_epoch_begin(self, epoch, logs=None):
+    def set_trainer(self, trainer):
```

Comment:
as all callbacks depend in trainer now, would it make sense to pass trainer in init...

Reply:
Yeah, I thought about it. But it seems like a bad idea because when the user creates a callback there is no trainer at the moment. I mean, now the user should do the following:
```
callback = Callback(some_callback_params)
trainer = Trainer(callback=callback)
```
And there will be problems if trainer should be used as a callback parameter.